### PR TITLE
feat: add support for in-memory client certificates

### PR DESCRIPTION
**Pull request by emilioalvap**
### Summary

Add support for clientCertificates from Playwright v1.47.0 by updating.

To note, code generation has been moved in https://github.com/microsoft/playwright/pull/32330. Some changes here adjust for that.

### CODE REVIEW

**Review by shahzad31 — APPROVED** [review]
LGTM !!

File: src/options.ts [hunk 1]
```diff
@@ -237,3 +245,32 @@ export function getCommonCommandOpts() {
...
+    }
+
+    // Revive serialized clientCertificates buffer objects
+    return (value ?? []).map(item => {
+      const revived = { ...item };
+      if (item.cert && !Buffer.isBuffer(item.cert)) {
+        revived.cert = parseAsBuffer(item.cert);
+      }
+      if (item.key && !Buffer.isBuffer(item.key)) {
+        revived.key = parseAsBuffer(item.key);
```
Comment (vigneshshanmugam) [inline src/options.ts:262]: we need to do the same for `pfx` option.

File: __tests__/options.test.ts [hunk 2]
```diff
@@ -148,4 +148,21 @@ describe('options', () => {
       },
     });
   });
+
+  it('cli parses playwrightOptions.clientCertificates correctly', async () => {
+    const test = {
+      clientCertificates: [
+        {
+          key: Buffer.from('This should be revived'),
```
Comment (vigneshshanmugam) [inline __tests__/options.test.ts:156]: Add a test for non-buffer values as we discussed.

**Review by vigneshshanmugam — COMMENTED** [review]
Code LGTM. One thing that can be improved is
- Add a validation during the push command for public locations and error out since cert paths, key paths wont work on the Service.

**Review by vigneshshanmugam — APPROVED** [review]

**Comment by vigneshshanmugam** [comment]
> Add a validation during the push command for public locations and error out since cert paths, key paths wont work on the Service.

@emilioalvap Are you planning to do this?

**Comment by emilioalvap** [comment]
@vigneshshanmugam Sure, I can add that to the PR

File: src/push/index.ts [hunk 3]
```diff
@@ -223,8 +223,16 @@ export function validateSettings(opts: PushOptions) {
   - CLI '--schedule <mins>'
   - Config file 'monitors.schedule' field`;
   } else if (opts.schedule && !ALLOWED_SCHEDULES.includes(opts.schedule)) {
-    reason = `Set default schedule(${opts.schedule
-      }) to one of the allowed values - ${ALLOWED_SCHEDULES.join(',')}`;
+    reason = `Set default schedule(${
+      opts.schedule
+    }) to one of the allowed values - ${ALLOWED_SCHEDULES.join(',')}`;
+  } else if (
+    opts.locations &&
```
Comment (vigneshshanmugam) [inline src/push/index.ts:230]:
```suggestion
    opts.locations.length > 0 &&
```

File: src/push/index.ts [hunk 4]
```diff
@@ -223,8 +223,16 @@ export function validateSettings(opts: PushOptions) {
...
+    (opts?.playwrightOptions?.clientCertificates ?? []).filter(cert => {
+      return cert.certPath || cert.keyPath || cert.pfxPath;
+    }).length > 0
+  ) {
+    reason = `Certificate path options (certPath, keyPath, pfxPath) are not supported on cloud locations, use in-memory alternatives (cert, key, pfx) when running on cloud.`;
```
Comment (vigneshshanmugam) [inline src/push/index.ts:235]:
```suggestion
    reason = `Certificate path options (certPath, keyPath, pfxPath) are not supported on globally managed locations, use in-memory alternatives (cert, key, pfx) when running on cloud.`;
```

**Review by vigneshshanmugam — APPROVED** [review]

**Pull request merged**